glean-core/ios/Glean/Glean.swift

```diff
@@ -114,19 +114,28 @@ public class Glean {
 
     /// Handle background event and send appropriate pings
     func handleBackgroundEvent() {
+        // This asks the OS for more time when going to background to perform critical tasks.
+        // This will be signaled complete when `sendPingsByName` is done.
         Glean.backgroundTaskId = UIApplication.shared.beginBackgroundTask(withName: "Glean Upload Task") {
```

Comment:
What happens when the app is moved to the background twice while the task is already running?

Reply:
According to the [docs](https://developer.apple.com/documentation/uikit/uiapplication/1623031-beginbackgroundtask) it's safe to call this multiple times as long as each execution has a matching call to the endBackgroundTask.  This can be called at any time, including when not yet going to background.  

So, to answer your question, it looks like we would have the background tasks executed twice, both of them requesting additional time to complete (if needed).  I think that this is the desired behavior we are looking for as each backgrounding event should launch the required pings.